### DISCUSSION TITLE
Version 0.1 release

### DISCUSSION THREAD

**Discussion by gwaybio** [cytomining/pycytominer#181] [General]
Hi community - @niranjchandrasekaran @shntnu @AnneCarpenter @bethac07 @staylorx @falquaddoomi to name a few.

I'd like to release the first official pycytominer version (with a 0.1 tag). Releasing this version will allow a smoother pip install experience, and will enable us to push to either conda-forge or bioconda. It will also mark a nice milestone that I have plans to build upon, and it is good practice for a publication.

I also think that the package is fairly stable, used fairly often, and that only a few more issues need resolution. In other words, in my view, we can release this version very soon (less than a day's work).

However, it's possible I am not seeing the full pictures. My question is therefore, do you see any other necessary items to accomplish before version 0.1 release? Possibly some functionality in the [`jump` branch](https://github.com/cytomining/pycytominer/tree/jump)? 

I've added the issues I know of here: https://github.com/cytomining/pycytominer/projects/1 They are up for grabs for folks to claim.

Note that releasing version 0.1 doesn't hinder any existing pipelines (as long as they're being installed via hash).

**Comment by shntnu** [comment]
> However, it's possible I am not seeing the full pictures. My question is therefore, do you see any other necessary items to accomplish before version 0.1 release? Possibly some functionality in the [`jump` branch](https://github.com/cytomining/pycytominer/tree/jump)?

I'm in favor of proceeding with the release and I note that @niranjchandrasekaran and @AnneCarpenter have give their 👍 

We needn't wait for the jump branch 

All the changes in that branch are related to a single file (effectively)
https://github.com/cytomining/pycytominer/pull/160/files

and that will very likely be moved out of pycytominer anyway
https://github.com/cytomining/pycytominer/pull/160#issuecomment-889005954

**Reply by gwaybio** [reply]
perfect. I'll wrap up the issues in https://github.com/cytomining/pycytominer/projects/1 and then proceed wit the release.

**Comment by staylorx** [comment]
Do you want to have pypi push actions in place before you release, or bring
those in with 0.2?

**Comment by gwaybio** [comment]
yeah, we could add that prior to version 0.1 - are you volunteering to set it up? (I haven't ever used them, nor know how much effort it takes 😄 )

I'm motivated to get this released to a stable version as a rotation student will be working on the workflow soon. So, if we can get pypi push actions up before Niranj can review #185 then let's do it!

**Reply by staylorx** [reply]
It's not a lot to do. Mostly legwork on your part. I can add the scripts, which will fail until the secrets are in place.

You'll need to create accounts on pypi, and I highly recommend creating an account on the test repo of pypi also. 

https://pypi.org/account/register/
https://test.pypi.org/account/register/

Once, you've got accounts, you'll get API tokens from both and create secret keys in the setup of this repo so you'll see something with `TEST_PYPI_API_TOKEN` and `PYPI_API_TOKEN` (you can name them whatever, but that's usually what I use):
![github actions secrets](https://user-images.githubusercontent.com/246923/152613247-a14e18a0-dc53-4fb4-9f3e-5bae5868023d.png)

I'll put in a PR with the action file once I run the builds a couple of times to see if it the branch builds wheels cleanly. That might be tomorrow at the rate I'm burning today.

**Reply by staylorx** [reply]
I can also create the accounts and tokens and push you the passwords if you like. Easy enough, but I don't want to take the fun away from anyone. :)

**Reply by gwaybio** [reply]
Done! Secrets added as well.

https://test.pypi.org/project/pycytominer/
https://pypi.org/project/pycytominer/

**Reply by staylorx** [reply]
I created PR #187 for the build and pypi publish. Work in progress as always; I hope it's a start. :)
